unity/src/main/java/io/delta/unity/metrics/UcCommitTelemetry.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.unity.metrics;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.delta.kernel.commit.CommitMetadata;
+import io.delta.kernel.internal.metrics.Timer;
+import java.util.Optional;
+
+/**
+ * Telemetry framework for Unity Catalog commit operations.
+ *
+ * <p>Collects timing metrics for commit operations and generates reports for successful and failed
+ * commits.
+ */
+public class UcCommitTelemetry {
```

Comment:
My goal here is to be VERY succint -- I want the metrics **collection** and metrics **result** and the overall **report** to succinctly be in one class